cub/test/catch2_test_device_scan_large_offsets.cu

```diff
@@ -0,0 +1,137 @@
+
+// TODO replace with DeviceScan interface once https://github.com/NVIDIA/cccl/issues/50 is addressed
+// Temporary wrapper that allows specializing the DeviceScan algorithm for different offset types
+template <typename InputIteratorT, typename OutputIteratorT, typename ScanOpT, typename InitValueT, typename OffsetT>
+CUB_RUNTIME_FUNCTION _CCCL_FORCEINLINE static cudaError_t dispatch_scan_wrapper(
+  void* d_temp_storage,
+  size_t& temp_storage_bytes,
+  InputIteratorT d_in,
+  OutputIteratorT d_out,
+  ScanOpT scan_op,
+  InitValueT init_value,
+  OffsetT num_items,
+  cudaStream_t stream = 0)
+{
+  using init_value_t = cub::detail::InputValue<InitValueT>;
+  init_value_t init_value_wrapper{init_value};
+
+  return cub::DispatchScan<InputIteratorT, OutputIteratorT, ScanOpT, init_value_t, OffsetT>::Dispatch(
+    d_temp_storage, temp_storage_bytes, d_in, d_out, scan_op, init_value_wrapper, num_items, stream);
+}
+
+DECLARE_LAUNCH_WRAPPER(dispatch_scan_wrapper, dispatch_exclusive_scan);
+
+// %PARAM% TEST_LAUNCH lid 0:1:2
+
+// List of offset types to be used for testing large number of items
+using offset_types = c2h::type_list<std::int64_t, std::uint64_t, std::int32_t, std::uint32_t>;
+
+template <typename ItemT>
+struct expected_sum_op
+{
+  uint64_t segment_size;
+
+  __host__ __device__ __forceinline__ ItemT operator()(const uint64_t index) const
+  {
+    uint64_t sum_per_full_segment = (segment_size * (segment_size - 1)) / 2;
+    uint64_t full_segments        = index / segment_size;
+    uint64_t index_within_segment = index % segment_size;
+
+    uint64_t sum_within_partial_segment = (index_within_segment * (index_within_segment - 1)) / 2;
+    uint64_t sum_over_full_segments     = full_segments * sum_per_full_segment;
+    return static_cast<ItemT>(sum_within_partial_segment + sum_over_full_segments);
+  }
+};
+
+template <typename ItemT>
+struct mod_op
+{
+  uint64_t segment_size;
+
+  __host__ __device__ __forceinline__ uint64_t operator()(const uint64_t index) const
+  {
+    return static_cast<ItemT>(index % segment_size);
+  }
+};
+
+CUB_TEST("DeviceScan works for very large number of items", "[scan][device]", offset_types)
+try
+{
+  using op_t     = cub::Sum;
+  using item_t   = std::uint32_t;
+  using index_t  = std::uint64_t;
+  using offset_t = typename c2h::get<0, TestType>;
+
+  // Clamp 64-bit offset type problem sizes to just slightly larger than 2^32 items
+  auto num_items_max_ull =
+    std::min(static_cast<std::size_t>(::cuda::std::numeric_limits<offset_t>::max()),
+             ::cuda::std::numeric_limits<std::uint32_t>::max() + static_cast<std::size_t>(2000000ULL));
+  offset_t num_items_max = static_cast<offset_t>(num_items_max_ull);
+  offset_t num_items_min =
+    num_items_max_ull > 10000 ? static_cast<offset_t>(num_items_max_ull - 10000ULL) : offset_t{0};
+  offset_t num_items = GENERATE_COPY(
+    values({
+      num_items_max,
+      static_cast<offset_t>(num_items_max - 1),
+      static_cast<offset_t>(1)
```

Comment:
question: I'm not sure why we are testing 1 item input. I thought this test is for large number of items. Is it a typo or am I missing something?

Reply:
This is to test if we would potentially run into underflows that become problematic for certain offset types. I think we can revisit this once we settled on a set of offset types. I'll add it to the TODO 👍